src/python/pants/backend/native/config/environment.py

```diff
@@ -109,10 +119,26 @@ def as_invocation_environment_dict(self):
 class Linker(datatype([
     'path_entries',
     'exe_filename',
-    'library_dirs',
+    'runtime_library_dirs',
     'linking_library_dirs',
     'extra_args',
-]), LinkerMixin): pass
+]), LinkerMixin):
+
+  @property
+  def with_tupled_collections(self):
```

Comment:
The 3 `with_tupled_collections` are unused - kill. Additionally, `with_*` reads like a verb and not like a noun, so either `@property` is not appropriate or the naming is suboptimal.